src/Node.ts

```diff
@@ -160,7 +160,7 @@ export abstract class Node {
   /** @lends Konva.Node.prototype */
   _clearCache(attr) {
     if (attr) {
-      delete this._cache[attr];
+      this._cache[attr] = undefined;
```

Comment:
May be better switch to `Map`?

Reply:
I think that test is better https://jsperf.com/map-vs-object-as-hashes/88

Probably we can switch to Map (need to check browser compatibility). But that is super micro-optimization. I don't think it will improve the performance of real apps.

Reply:
On latest Chrome case 88 still much faster: `864,097,105 ops` (Map#get) and `33,567,486 ops` (get (Object)). it's 25 times faster. Also this test doesn't take into account (Map#set) and (Map#delete) which I expect could be much better than `delete _cache[key]` becaouse last out always allocate new object and `__proto__`. Also this test operate only on small set pre-generated set and speed could be much better on large dynamic set.

[About compatibility](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Map#%D0%A1%D0%BE%D0%B2%D0%BC%D0%B5%D1%81%D1%82%D0%B8%D0%BC%D0%BE%D1%81%D1%82%D1%8C_%D1%81_%D0%B1%D1%80%D0%B0%D1%83%D0%B7%D0%B5%D1%80%D0%B0%D0%BC%D0%B8). With also everything ok. Btw typescript could polyfill Map/Set if it not supported by prtty old browsers in target set as ES5

Reply:
I agree that is it much faster. Just want to mention that it probably will not affect the performance in general, because there are o lot of other much slower parts of Konva (like canvas rendering itself). So that 25x boost will be just a 0.5% boost in the real canvas app.

But anyway, I am happy to see a pull request for that change if anyone is interested.

Reply:
Interesting, tried to create a test without `__proto__`. Result is almost the same.
https://jsperf.com/map-vs-object-as-hashes/89

Reply:
I think, using `Map` would be a good improvement, going forward. I agree with @MaxGraey.

@lavrton, for a general-purpose cache usage, you are totally right, keys would have to be deleted still. The current set of cached keys in Konva is more restricted and mostly the same for each Node specialization; that was the reasoning behind my suggestion. You are also correct in terms of no / minor performance difference for most cases.

Okay, let me close this specific request. And regarding `Map` – depending on how much backward compatibility we want / don't want to preserve for Konva, going forward. I may be able to get to that. My own experiences with `Map` are quite good so far.